### PR TITLE
Start the cluster operator.KubeCluster when created asynchronously

### LINKED ISSUE #598: Creating KubeCluster asynchronously does not start a cluster
**Describe the issue**:

When creating the `operator.KubeCluster` with the `asynchronous` param to `True`, the cluster is never started.

In the example below, I got this error when I try to call the `adapt` method:
```json
{"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"daskclusters.kubernetes.dask.org \"dask-john-b1b26482-b\" not found","reason":"NotFound","details":{"name":"dask-john-b1b26482-b","group":"kubernetes.dask.org","kind":"daskclusters"},"code":404}
```

My guess is that the `_start` method of `operator.KubeCluster` is never called. With the `classic.KubeCluster`, when running asynchronously, the `_start` method was called in the `distributed.SpecCluster` class [here](https://github.com/dask/distributed/blob/main/distributed/deploy/spec.py#L395:L403). The new `operator.KubeCluster` inherits from `distributed.Cluster` instead and if I add the  `__await__(self):` it works and the cluster is created.

**Minimal Complete Verifiable Example**:

```python
import asyncio
from dask_kubernetes.operator.kubecluster.kubecluster import KubeCluster
from dask.distributed import Client
import time


def say_hello():
    print("hello from worker!")
    time.sleep(10)


async def try_cluster():
    cluster = await KubeCluster(
        image="ghcr.io/dask/dask",
        asynchronous=True,
    )
    await cluster.adapt(1, 1)

    client = Client(
        cluster, asynchronous=True
    )
    await client.submit(say_hello)
    await cluster.close()


if __name__ == '__main__':
    loop = asyncio.get_event_loop()
    coroutine = try_cluster()
    loop.run_until_complete(coroutine)
```

**Anything else we need to know?**:

I opened a question on the Dask's forum [here](https://dask.discourse.group/t/new-operator-kubecluster-and-speccluster/1212) and this bug was found when trying to run the new `operator.KubeCluster` (as recommended by Dask's documentation) with prefect (opened issue [here](https://github.com/PrefectHQ/prefect-dask/issues/38)).

**Environment**:

- Dask version: 2022.10.2
- Python version: 3.8.13
- Operating System: Ubuntu 20.04.5 LTS
- Install method (conda, pip, source): pip

### PR DESCRIPTION
**Pull request by john-jam**
## Description

This PR adds support to automatically start an `operator.KubeCluster` when created with the `asynchronous` parameter set to `True`.

## References

- Closes #598

### CODE REVIEW

**Comment by john-jam** [comment]
The test I added [was failing](https://github.com/dask/dask-kubernetes/actions/runs/3403409505/jobs/5659919951#step:5:189) (`test_kubecluster_async`) and I think it's because of the `test_discovery` that is also failing. When I remove it, my test succeed.

I tried [using 2 async clusters](https://github.com/dask/dask-kubernetes/pull/602/files#diff-aa4cf4a8db941ee408ec1be7480c4c3f457d5943ff5c9ecb1bb147ab9d591c10R9) in the `test_discovery` and the errors regarding the event loops don't show up anymore. Also, I had to wrap the `k8s` API delete cluster call in a try catch since this test uses 2 instances of `KubeCluster` pointing to the same underlying `k8s` dask cluster.

I am not an expert with event loops and maybe those fixes are not the scope of this PR. I can remove my updates in [this commit](https://github.com/dask/dask-kubernetes/pull/602/commits/10ce587af63d7406e9e98eba5364500112f5f5af) if you like. What do you think @jacobtomlinson ?

**Review by jacobtomlinson — COMMENTED** [review]
Thanks for this @john-jam.

The asyncio implementation here may not be in the best state. I wonder if @graingert has any thoughts?

**Comment by john-jam** [comment]
Any updates on this @jacobtomlinson 
If there's something to update/remove in this PR, please let me know!

File: dask_kubernetes/operator/kubecluster/kubecluster.py [hunk 1]
```diff
@@ -543,13 +552,20 @@ async def _close(self, timeout=None):
...
+                    await custom_objects_api.delete_namespaced_custom_object(
+                        group="kubernetes.dask.org",
+                        version="v1",
+                        plural="daskclusters",
+                        namespace=self.namespace,
+                        name=self.name,
+                    )
+                except ApiException as e:
+                    if e.reason == "Not Found":
+                        # TODO: maybe add a warning or do nothing?
```
Comment (jacobtomlinson) [inline dask_kubernetes/operator/kubecluster/kubecluster.py:565]: Let's add a warning here. Something like `Failed to delete DaskCluster, looks like it has already been deleted.`.

**Review by jacobtomlinson — COMMENTED** [review]
Sorry for letting this hang. One small comment. Could you also pull in from `main` so that CI gets into a good state. Will squash on merge anyway.

**Comment by john-jam** [comment]
The PR has been rebased.

Also, I [removed the test skip](https://github.com/dask/dask-kubernetes/pull/602/commits/db5a8992b4b9e4d8d8f168295ffc552b1d061a55) for `test_cluster_from_name` since I think the race condition was the cluster deletion (called twice on each instance pointing to the same cluster).
We can see our new warning called in one instance when we run this test:
```
[0] $ export KUBECONFIG=.pytest-kind/pytest-kind/kubeconfig && pytest -o log_cli=true dask_kubernetes/operator/kubecluster/tests/test_kubecluster.py::test_cluster_from_name
============================================================================================================================ test session starts =============================================================================================================================
platform linux -- Python 3.8.13, pytest-7.2.0, pluggy-1.0.0 -- /home/john/projects/github.com/john-jam/dask-kubernetes/venv/bin/python
cachedir: .pytest_cache
rootdir: /home/john/projects/github.com/john-jam/dask-kubernetes, configfile: setup.cfg
plugins: timeout-2.1.0, rerunfailures-10.2, asyncio-0.20.1, kind-22.9.0
timeout: 300.0s
timeout method: signal
timeout func_only: False
asyncio: mode=strict
collected 1 item                                                                                                                                                                                                                                                             

dask_kubernetes/operator/kubecluster/tests/test_kubecluster.py::test_cluster_from_name 
------------------------------------------------------------------------------------------------------------------------------- live log call --------------------------------------------------------------------------------------------------------------------------------
DEBUG    kopf._core.reactor.running:running.py:499 Starting Kopf 1.35.6.
DEBUG    kopf.activities.startup:execution.py:271 Activity 'noop_startup' is invoked.

...
WARNING  dask_kubernetes.operator.kubecluster.kubecluster:kubecluster.py:575 Failed to delete DaskCluster, looks like it has already been deleted.
...
================================================================================================================== 1 passed, 4 warnings in 73.08s (0:01:13) ==================================================================================================================
```

**Comment by john-jam** [comment]
Hi @jacobtomlinson, any remaining items I could work on to merge this one?

**Comment by TheBeege** [comment]
@graingert , considering that @jacobtomlinson is away, can you help out with this at all? Is there any remaining work to get this merged?

**Comment by jacobtomlinson** [comment]
Sorry I'm back now. Thanks for fixing things up. Lets go ahead!

**Pull request merged**